src/bp-core/bp-core-functions.php

```diff
@@ -4937,7 +4931,7 @@ function( $file ) {
 		$latest_deprecated_functions_versions = array_splice( $latest_deprecated_functions_versions, -2 );
 	}
 
-	$index_initial_version = array_search( $first_major, $latest_deprecated_functions_versions, true );
+	$index_initial_version = array_search( $initial_version, $latest_deprecated_functions_versions, true );
 	if ( false !== $index_initial_version ) {
```

Comment:
I believe, this is going to be a problem because, **`$index_initial_version`** is most likely going to be an integer value of 0 (zero), if I'm not mistaken.

Reply:
I think it's ok, 0 or another integer value means array contains the initial version so we can remove it.